### ISSUE TITLE
Declaring a regex while using useHead and a class on the top div of a page causes it to crash

### ISSUE THREAD

**Issue by Predota** [nuxt/nuxt#22425]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.6.5
- Nitro Version:    2.5.2
- Package Manager:  npm@9.4.2
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-bbxqum?file=pages%2Findex.vue

### Describe the bug

The page is crashing when :
- I declare a regex
- I am using useHead
- I have a class on the top div of the page

Removing one of the three fixes the issue. 

Using new Regex() to declare the regex also fixes the issue.

### Logs

```shell-script
500
useHead is not defined

at _sfc_main.setup (./pages/index.js:31:23)
at callWithErrorHandling (./node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:156:18)
at setupStatefulComponent (./node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:7190:25)
at setupComponent (./node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:7151:36)
at renderComponentVNode (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:614:15)
at renderVNode (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:743:14)
at renderComponentSubTree (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:698:7)
at renderComponentVNode (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:631:12)
at renderVNode (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:743:14)
```

**Comment by danielroe** [comment]
cc: @antfu

**Comment by danielroe** [comment]
Interesting!

**Comment by danielroe** [comment]
Here's a smaller repro:

```vue
<script setup>
const passwordRegex = /`/

ref()
</script>

<template>
  <div class="">Hello</div>
</template>
```

**Comment by danielroe** [comment]
Checking, it seems to be an unimport issue, but only on server side. (Import is added on client-side build.)

**Comment by antfu** [comment]
The fix should be shipped in unimport v3.1.1. Before the next Nuxt release, try to remove your lock and do a clean install to have the latest unimport. Thanks.

**Comment by Predota** [comment]
Hey ! Thank you all for the quick fix and for all the work you put in open source projects !
